plot/plotcore/metadata.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2024, Cogent Core. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotcore
+
+import (
+	"cogentcore.org/core/base/errors"
+	"cogentcore.org/core/base/metadata"
+	"cogentcore.org/core/tensor/table"
+)
+
+// todo: make a more systematic version driven by reflect fields
+// key issue is the same nullable type issue: you don't want to set everything
+// in md.
+
+// also the Column name and IsString should not be in the struct!
+
+// fromMeta sets plot options from meta data.
+func (po *PlotOptions) fromMeta(dt *table.Table) {
+	po.FromMetaMap(dt.Meta)
+}
+
+// FromMetaMap sets plot options from meta data map.
+func (po *PlotOptions) FromMetaMap(meta metadata.Data) {
+	if typ, err := metadata.Get[string](meta, "Type"); err == nil {
```

Comment:
You could write a nice DRY helper function for all of this stuff...

Reply:
next time.

Reply:
will do later.